server/src/com/cloud/api/ApiServer.java

```diff
@@ -264,10 +266,11 @@ public void handleAsyncJobPublishEvent(String subject, String senderAddress, Obj
         String info = job.getCmdInfo();
         String cmdEventType = "unknown";
         if (info != null) {
-            String marker = "\"cmdEventType\"";
-            int begin = info.indexOf(marker);
-            if (begin >= 0) {
-                cmdEventType = info.substring(begin + marker.length() + 2, info.indexOf(",", begin) - 1);
+            Type type = new TypeToken<Map<String, Object>>(){}.getType();
+            Map<String, Object> cmdInfo = ApiGsonHelper.getBuilder().create().fromJson(info, type);
```

Comment:
is this Object necessary as the mapped type? Couldn't it be String directly? If you are using the api solely for collecting the cmdEventType, you could build the objects as String already and the typecheck on line 272 would be unnecessary.